### PR TITLE
Add searchgui datatype

### PR DESCRIPTION
**Pull request by jj-umn**
Need to include searchgui version in metadata of searchgui_archive in order to associate witth correct version of peptideshaker.

### CODE REVIEW

File: config/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -179,7 +179,7 @@
     <datatype extension="sf3" type="galaxy.datatypes.proteomics:Sf3" display_in_upload="true" />
     <datatype extension="cps" type="galaxy.datatypes.binary:Binary" subclass="True" display_in_upload="true" />
     <datatype extension="ct" type="galaxy.datatypes.tabular:ConnectivityTable" display_in_upload="True"/>
-    <datatype extension="searchgui_archive" type="galaxy.datatypes.binary:CompressedArchive" subclass="True" display_in_upload="True"/>
+    <<datatype extension="searchgui_archive" type="galaxy.datatypes.binary:SearchGuiArchive" display_in_upload="True"/>
```
Comment (bgruening) [inline config/datatypes_conf.xml.sample]: `<<` This looks like a typo @jj-umn.

**Comment by bgruening** [comment]
@jj-umn looks good to me.
Can you please fix the pep8 errors pointed out by travis?

```
./lib/galaxy/datatypes/binary.py:1056:1: W293 blank line contains whitespace
./lib/galaxy/datatypes/binary.py:1057:1: E302 expected 2 blank lines, found 1
./lib/galaxy/datatypes/binary.py:1064:5: E301 expected 1 blank line, found 0
./lib/galaxy/datatypes/binary.py:1083:15: E271 multiple spaces after keyword
./lib/galaxy/datatypes/binary.py:1107:1: W293 blank line contains whitespace
./lib/galaxy/datatypes/binary.py:1107:1: W391 blank line at end of file
```

**Comment by jj-umn** [comment]
Error seems to involve DB versions.   Unrelated to this pull request?

Console:

LOG:  autovacuum launcher started
Traceback (most recent call last):
  File "./scripts/manage_db.py", line 33, in <module>
    invoke_migrate_main()
  File "./scripts/manage_db.py", line 30, in invoke_migrate_main
    main( repository=repo, url=db_url )
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/shell.py", line 209, in main
    ret = command_func(*_kwargs)
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/api.py", line 186, in upgrade
    return _migrate(url, repository, version, upgrade=True, err=err, *_opts)
  File "<string>", line 2, in _migrate
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/util/__init__.py", line 160, in with_engine
    return f(_a, *_kw)
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/api.py", line 345, in _migrate
    changeset = schema.changeset(version)
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/schema.py", line 82, in changeset
    changeset = self.repository.changeset(database, start_ver, version)
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/repository.py", line 225, in changeset
    changes = [self.version(v).script(database, op) for v in versions]
  File "/galaxy/eggs/sqlalchemy_migrate-0.9.6-py2.7.egg/migrate/versioning/version.py", line 204, in script
    "There is no script for %d version" % self.version
AssertionError: There is no script for 130 version
Build step 'Execute shell' marked build as failure

**Comment by natefoo** [comment]
@jj-umn I don't think jenkins is expected to work at this point.
@martenson We're not doing a fresh clone every time? Because this happens if `lib/galaxy/model/migrate/versions/*.pyc` exist but their corresponding `.py` do not, which can happens if you downgrade.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
@natefoo you are correct, that is why this was a problem, not anymore though

@jj-umn correct, unrelated to this PR, sorry that Jenkins bothered you :)

the two sets of tests that Jenkins runs for us are quite reliable now, of course there will be a glitch with the tests now and then that is not PR-related but I think that the results are worth checking

I will send an email to commiters soon about Jenkins testing

**Comment by martenson** [comment]
Thank  you @jj-umn !

**Pull request merged**